### PR TITLE
TRUNK-6049: Migrate tracking of Logged-In Users into core

### PR DESCRIPTION
**Pull request by jnsereko**
## Description of what I changed

migrated CurrentUsers.java class into core based on the discussion on talk https://talk.openmrs.org/t/view-logged-in-users-is-empty/19348/15

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-6049

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by jnsereko** [comment]
now that we have this logic migrated here, i think we should have https://github.com/openmrs/openmrs-module-adminui/pull/64 merged if it matches the requirements

**Comment by jnsereko** [comment]
> @jnsereko this is great! kindly fix the link to the ticket you worked.

thanks @jwnasambu

File: web/src/main/java/org/openmrs/web/SessionListener.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.web.user.CurrentUsers;
+
+/**
+ * Handles events of session life cycle. <br>
+ * <br>
+ * This is set by the web.xml class
+ */
```
Comment (dkayiwa) [inline web/src/main/java/org/openmrs/web/SessionListener.java:23]: https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-NewPublicMethodsandClasses

File: web/src/main/java/org/openmrs/web/user/CurrentUsers.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+import org.apache.commons.lang.StringUtils;
+import org.openmrs.User;
+import org.openmrs.web.WebConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility methods which maintain the list of current users.
+ */
```
Comment (dkayiwa) [inline web/src/main/java/org/openmrs/web/user/CurrentUsers.java:30]: https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-NewPublicMethodsandClasses

**Comment by dkayiwa** [comment]
@jnsereko how do i test these changes as an end user?

File: web/src/main/java/org/openmrs/web/SessionListener.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Handles events of session life cycle. <br>
+ * <br>
+ * This is set by the web.xml class
+ * @since 2.6.0
+ */
+public class SessionListener implements HttpSessionListener {
+	
+	protected final Log log = LogFactory.getLog(getClass());
```
Comment (dkayiwa) [inline web/src/main/java/org/openmrs/web/SessionListener.java:27]: https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-HowToLog

Comment (jnsereko) [reply web/src/main/java/org/openmrs/web/SessionListener.java:27]: https://user-images.githubusercontent.com/58003327/162873525-4c64bf41-f9b2-45ca-bb89-f536d1968b05.mp4

**Comment by jnsereko** [comment]
hello @dkayiwa 
This has taken some time but i also depends on https://github.com/openmrs/openmrs-module-adminui/pull/64 
Please kindly check it out

**Comment by dkayiwa** [comment]
Did you remove the necessary classes from the legacyui module and try to run it on the openmrs platform with these changes?

**Comment by rukundob451** [comment]
Hello there, I am seeing that some checks were not successful such as the CodeQL/Analyze(java) any recommendations to that?
@jnsereko and @dkayiwa

**Comment by dkayiwa** [comment]
@rukundob451 you can ignore the CodeQL check

**Pull request closed without merging**